admin/src/components/ReactIconsSelector/index.tsx

```diff
@@ -122,49 +138,93 @@ const ReactIconsSelector: React.FC<IReactIconsSelector> = ({
         required={required}
         error={error}
         startAction={
-          <FieldAction onClick={toggleModal}>
-            {value ? <IconComponent icon={value} /> : <ReactIcons.TbSearch />}
+          <FieldAction onClick={toggleModal} style={inputButtonStyle} left="0">
+            {value ? (
+              <IconComponent icon={value} />
+            ) : (
+              <ReactIcons.TbSearch width="1.25rem" height="1.25rem" />
+            )}
           </FieldAction>
         }
         endAction={
           !!value && (
-            <FieldAction onClick={() => changeIcon("")}>
+            <FieldAction
+              onClick={() => changeIcon("")}
+              style={inputButtonStyle}
+              right="0"
+            >
               <ReactIcons.TbX />
             </FieldAction>
           )
         }
       />
 
       {isModalVisible && (
-        <ModalLayout onClose={toggleModal} labelledBy="title">
+        <ModalLayout
+          onClose={toggleModal}
+          labelledBy="title"
+          width="65vw !important"
+        >
           <ModalHeader>
             <Typography fontWeight="bold" id="title">
               Select icon
             </Typography>
           </ModalHeader>
-          <ModalBody>
-            <Flex direction="row" wrap="wrap" gap={2}>
-              {selectedIconLibrary ? (
-                selectableIcons.length > 0 ? (
-                  selectableIcons.map((icon) => (
-                    <Box
-                      key={icon}
-                      variant="secondary"
-                      onClick={() => {
-                        toggleModal();
-                        changeIcon(icon);
-                      }}
-                    >
-                      <IconComponent size={30} icon={icon} />
-                    </Box>
-                  ))
-                ) : (
-                  <Typography variant="pi">No icons available</Typography>
-                )
-              ) : (
-                <Typography variant="pi">No icon library selected</Typography>
-              )}
-            </Flex>
+          <ModalBody
+            minHeight="70vh"
+            maxHeight="unset !important"
+            overflow="hidden"
+          >
+            <TextInput
```

Comment:
The search field extends the scroll bar of the icon list:
![grafik](https://github.com/DanielPantle/strapi-plugin-react-icons/assets/34894514/a26130b3-e524-4cee-8ff5-f7506ba188e6)
I think this design can be improved by adapting the width.

Also please think about using the [search bar component from strapi](https://design-system-git-main-strapijs.vercel.app/?path=/docs/design-system-components-searchbar--base.). Like this we have a automatic "x" button when something is typed in.

Reply:
@DanielPantle Replaced with Search bar! Thanks again for this advice, I really enjoy this type of stuff and want to get better :)